monitor/metrics/checker.go

```diff
@@ -26,6 +27,9 @@ type Checker struct {
 	alertCh   chan *api.Alert
 	metrics   *Store
 	threshold float64
+
+	failedPeersMu sync.Mutex
+	failedPeers   map[peer.ID]bool
```

Comment:
`map[peer.ID]struct{}` ?

Reply:
actually, `map[peer.ID]int` and record how many times something has alerted, and then make the threshold of number of times something should alert at most a variable at the top?